lib/vintage_net_qmi/mcc_mnc.ex

```diff
@@ -0,0 +1,33 @@
+defmodule VintageNetQMI.MCCMNC do
+  @moduledoc """
+  Automatically generated file based on https://s3.amazonaws.com/mcc-mnc.net/mcc-mnc.csv
+  """
+  @external_resource "lib/mcc-mnc.csv"
+  @csv_data File.read!("lib/mcc-mnc.csv")
+            |> String.trim()
+            |> String.split("\r\n")
+            |> Enum.map(&String.split(&1, ";"))
+            |> tl()
+
+  @providers @csv_data
+             |> Enum.map(fn [mcc, mnc, _, _, _, _, _, pro, _, _ | _] -> {mcc, mnc, pro} end)
+             |> Enum.uniq_by(fn {mcc, mnc, _} -> {mcc, mnc} end)
+
+  @doc """
+  Get the provider name for a network id.
+
+  ## Examples
+
+        iex> VintageNetQMI.MCCMNC.provider("310", "410")
+        "AT&T"
+
+        iex> VintageNetQMI.MCCMNC.provider("000", "000")
+        nil
+  """
+  @spec provider(String.t(), String.t()) :: String.t() | nil
+  for {mcc, mnc, provider} <- @providers do
+    def provider(unquote(mcc), unquote(mnc)), do: unquote(provider)
```

Comment:
Can this be done in a way that makes the resulting `.beam` smaller?

While it's not large, it feels a little silly to have such a lightly used database in DRAM all the time. Here are some ideas:
1. Figure out a way to make it small enough to not care. I did a quick test for making making the key be an integer representation of the mcc/mnc. That reduced the compressed .beam file from 79452 to 53472. Maybe there's a way to do even better. I didn't look if the string values were highly repetitive and Erlang wasn't deduplicating them.
2. Keep the .csv file or a processed version of it in the `priv` directory. Look the provider up when needed.

Reply:
updated it to just do the lookup at runtime. keepin it simple